server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -221,12 +176,13 @@ private static void addCommand(StringBuilder builder, String name, int arity, in
 
    }
 
-   private static void handleLongResult(ChannelHandlerContext ctx, Long result) {
-      ctx.writeAndFlush(RespRequestHandler.stringToByteBuf(":" + result + "\r\n", ctx.alloc()), ctx.voidPromise());
+   protected static void handleLongResult(Long result, ByteBufPool alloc) {
+      // TODO: this can be optimized to avoid the String allocation
+      stringToByteBuf(":" + result + "\r\n", alloc);
    }
 
-   static void handleThrowable(ChannelHandlerContext ctx, Throwable t) {
-      ctx.writeAndFlush(RespRequestHandler.stringToByteBuf("-ERR" + t.getMessage() + "\r\n", ctx.alloc()), ctx.voidPromise());
+   protected static void handleThrowable(ByteBufPool alloc, Throwable t) {
+      stringToByteBuf("-ERR" + t.getMessage() + "\r\n", alloc);
```

Comment:
Just realized now, missing the space between `-ERR` and the exception message.

Reply:
Ah, good catch. It has been like that for a while now. Fixed.